crates/sui-core/src/unit_tests/authority_tests.rs

```diff
@@ -771,6 +771,48 @@ async fn test_publish_non_existing_dependent_module() {
     );
 }
 
+// make sure that publishing a package above the size limit fails
+#[tokio::test]
+async fn test_package_size_limit() {
+    let (sender, sender_key): (_, AccountKeyPair) = get_key_pair();
+    let gas_payment_object_id = ObjectID::random();
+    let gas_payment_object =
+        Object::with_id_owner_gas_for_testing(gas_payment_object_id, sender, u64::MAX);
+    let gas_payment_object_ref = gas_payment_object.compute_object_reference();
+    let mut package = Vec::new();
+    let mut package_size = 0;
+    // create a package larger than the max size
+    while package_size <= MAX_MOVE_PACKAGE_SIZE {
+        let mut module = file_format::empty_module();
+        // generate unique name
+        module.identifiers[0] = Identifier::new(format!("TestModule{:?}", package_size)).unwrap();
+        let module_bytes = {
+            let mut bytes = Vec::new();
+            module.serialize(&mut bytes).unwrap();
+            bytes
+        };
+        package_size += module_bytes.len();
+        package.push(module_bytes);
+    }
+    // the published object size will be larger because the "module rewriter" adds a 20 byte self address to each moudle
```

Comment:
dumb q and a nit: is `"module rewriter"` reliable so that it's safe to assume 20 bytes? is it simpler to just update the above condition to 
```
while package_size < MAX_MOVE_PACKAGE_SIZE {
```

Reply:
Fortunately, Ashok got rid of the rewriter upstream, so I deleted this confusing code/comment!